### PR TITLE
Fix checking version of PHP

### PR DESCRIPTION
**Pull request by avil13**
Stack didn't resolve and came to check version of php

### CODE REVIEW

**Comment by damieng** [comment]
I don't understand why the setTimeout is necessary - the exit event should be emitted at the right time.

Do you have a repro of where the problem occurs and on which operating system?

**Comment by avil13** [comment]
I have a version of php 7.1.
But when doing the check, the program did not have time to get the version number.
To wait for the output from the stack I used _setTimeout_.
It worked for me at that moment.
But now the plug-in stopped working at all.
Mac Pro OS With El Capitan

**Comment by Arcanemagus** [comment]
> But when doing the check, the program did not have time to get the version number.

@damieng it might be useful to disable usage of the configuration files (`-n`) when grabbing the version, it can be quite an expensive operation to load all the modules specified in the configuration files. Is there any reason this isn't just using `php --version` and parsing the output?

**Comment by damieng** [comment]
I thought parsing the output would be less reliable but if there's a
non-negligible delay (1s+) I'd take a PR to switch it

File: lib/main.js [hunk 1]
```diff
@@ -66,28 +66,33 @@ class PHPLanguageClient extends AutoLanguageClient {
...
-          }
-        } else {
-          atom.notifications.addError('IDE-PHP encounted an error using the PHP runtime.', {
-            dismissable: true,
-            description: stdErr != null ? `<code>${stdErr}</code>` : `Exit code ${exitCode}`
-          })
-          reject()
-        }
+          setTimeout(() => {
+            if (exitCode === 0 && stdOut != null) {
```
Comment (lachieh) [inline lib/main.js]: `stdOut != null` returns true here. Better to use `!stdOut` which is falsey.

**Comment by lachieh** [comment]
This solves the issue for me

**Comment by damieng** [comment]
Went with closed event instead of setTimeout as former should be more reliable.

**Comment by inoas** [comment]
For me the fix was to run `which php` and the path as PHP Path in ide-php settings in atom
Version: MacOS 10.12.6; Atom 1.21.0; ide-php 0.6.9

**Pull request closed without merging**